added mxcube video streamer support

mockoocy opened the pull request:
Adds [mxcube video-streamer](https://github.com/mxcube/video-streamer/tree/main)

It will be started by mxcubecore, as a subprocess, similarly to [this](https://github.com/mxcube/mxcubecore/blob/develop/mxcubecore/HardwareObjects/mockup/MDCameraMockup.py)

It also required change to the mocked MD3 video device server. Video streamer works under assumption that `video_last_image_counter` changes independently of wether the image has been polled or not. Currently the counter is increment only when we poll the image.

elmjag commented:
I have tested this, and took liberty to make some changes:

https://github.com/elmjag/micromax/commits/add-video-streamer
https://github.com/elmjag/micromax/commit/11c1960899d53122bcba825fcb17eaffd614bb1e

I changed slighty the style of the code implementing `video_last_image_counter` attribute in` b-micromax-md3-pc/md3video.py`.

I removed the code that clones `mxcube-video-streamer` repository in MXCuBE image. The mxcube-video-streamer package is already installed, via dependency to mxcubeweb package. The reason we clone `mxcubeweb` and `mxcubecore` repositories, is because we have forked them. I hope we don't need to fork mxcube-video-streamer, but just use the upstream version as-is.

These changes work fine. I'll merge https://github.com/elmjag/micromax/commits/add-video-streamer in the comming days, unless you object to my changes!

fabcor-maxiv commented on mxcube/Dockerfile:
```diff
@@ -5,6 +5,8 @@ RUN apt-get update
     # need for lucid3
     libgl1-mesa-glx \
+    # for video streamer
+    ffmpeg \
```
You could also add it as a *conda* dependency.

This is what is going to happen upstream: <https://github.com/mxcube/mxcubeweb/pull/1504>.

mockoocy replied:
Done

mockoocy committed:
added mxcube video streamer support
Emulate 'video_last_image_counter' attribute of the MD3Up video
tango device. The video-streamer depends on that attribute.

Include 'ffmpeg' package into MXCuBE image, as video-streamer
requires that package.

mockoocy committed:
include video-streamer in circus.conf

mockoocy commented:
new changes: added `video-streamer` to `circus.conf` and a `docker-compose` volume to mount a config file for it.

mockoocy committed:
add video streamer port in compose

elmjag commented:
I have in a round-about-way merged these changes: https://github.com/elmjag/micromax/commit/183aecab19c90f5afdc92bc8362b32e51d95d1b7

I modified your PR a bit. The biggest difference is that the video-streamer config file is managed by the dashboard. We also changed the video streaming port to 8082, to avoid using a 'common' port number 8000.

As far as I can tell, the MPEG streaming works fine now in the emulator. Let me know if I have missed something.
